cli/azd/pkg/apphost/generate.go

```diff
@@ -378,6 +380,12 @@ func (b *infraGenerator) LoadManifest(m *Manifest) error {
 			b.connectionStrings[name] = *comp.ConnectionString
 		}
 
+		if comp.Type == "value.v0" && comp.Value != "" && !hasInputs(comp.Value) {
```

Comment:
Hard for me to have a complete picture of what's going on here - but I am worried this will be a little fragile.  My read here is we want to have two cases:

1. Where the `.Value` property is a fixed value, with no expressions in the string (e.g. `"foo"`)
2. Where the `.Value` property is an expression, which may reference inputs (e.g `"{foo.inputs.bar}"`)

What I am wondering is would there ever be a case where `value.v0` might refer to something other than `inputs`?  It feels like a general purpose way to model fixed values (and values that are formed from expressions) and scoping it to just inputs here may break subtly down the line.

Is this really specific to just inputs or could we reference other properties of resources?  From the code it seems like it's possible that we might only ever reference inputs to the value itself?  Just trying to puzzle out why we also have to add an input parameter based on a `value.v0` in this case - would it not already be refercing a `parameter.v0`?

Reply:
value.v0 can define its content from:
- connectionString
- value

In both cases, the content can be, either:
- a string constant, like "foo"
- a reference to other resource or to its own inputs, like `{otherResource.outputs.foo}` or `{thisResource.inputs.bar}`
- a complex expression combining the 2 previous options, like `making {otherResource.outputs.foo} all {thisResource.inputs.bar} complicated`

The difference between a parameter.v0 and input.v0
- parameter is not expected to use the `connectionString` field (so far, lol)
- there is a way for customers to create parameters in their AppHost program by using some API, like `builder.AddParameter(...)`
- value.v0 is more like an internal way to link parent-child resources without using the `parent` field.  For example, a previous resource: `database` + `parent=cosmos.server` is not represented as a `value.v0` with a complex expression like, `{server.outputs.connectionString};database=foo;` in its `connectionString` 
- Customers can't create `value.v0` from the program, it is the way Aspire maps things together in the manifest.

Reply:
Not valid.
We are currently returning a `not supported error` if a resource tries to reference `outputs` from another resource.
The rule is that resources can uses inputs from its own only.

Reply:
validating `using inputs from its own`:
https://github.com/Azure/azure-dev/blob/main/cli/azd/pkg/apphost/generate.go#L520

If another resource needs the same input, it must reference the parent's resource (either from a parameter or from some other resource)

Reply:
Gotch - makes sense - thanks for the detailed explanation!